Turbine_single_tube.m

```diff
@@ -0,0 +1,95 @@
+% |              University of Naples Federico II.                                             |
+% |Version     : 2.1                                                                           |
+% |Date        : 05/06/2020                                                                    |
+% |Modified    : 15/01/2021                                                                    |
+% |Description : characteristic curves evaluation for a Darrieus turbine through the           |
+%                 multiple streamtubes theory                                                  |
+% |Reference   : Tognaccini R., (2019), "Lezioni di Aerodinamica dell'ala rotante"             |
+%                De Vries O., (1979), "Fluid Dynamic Aspects of Wind Energy Conversion"        |
+% |Input       :  (c) = chord of the blade                                                     |
+%                 (R) = blade radius                                                           |
+%                 (Cla) = slope coefficient of the Cl curve for the considered airfoil         |
+%                 (N) = number of blades                                                       |
+%                 (Cd) = drag coefficient of the considered airfoil                            |
+%                                                                                              |
+% |Output      :  (cp) = vector of power coefficient values for the                            |
+%                 turbine                                                                      |
+%                 (cq) = vector of torque coefficient values for the                           |
+%                 turbine                                                                      |
+%                 (lambda) = vector of tip speed values corresponding to                       |
+%                 the cp and cq coefficients                                                   |
+% |Note        :  The function works                                                           |
+% ==============================================================================================
+function [Cp, Lambda, Cq] = Turbine_single_tube(Cla, c, R, N, cd, alphamax)
+
+%Domain vector, initial values. 
+y1=linspace(0,100,100);       %Introduction of the variable y
+phiv = linspace(0,360,100);     %Phi domain (deg)
+phiv = deg2rad(phiv);           %Phi domain (rad)  
+alphamax = deg2rad(alphamax);
+%Equations 
+ 
+for i = 1:length(y1)            % a, lambda, cp, cq computing 
+    
+    y=y1(i);
+    
+%   Computing of the induction factor a
+    Y= ((y-sin(phiv)).^2+cos(phiv).^2).*Cla.*atan(cos(phiv)./(y+sin(phiv))).*...
+        cos(phiv+atan(cos(phiv)./(y+sin(phiv))));   
+    A =   trapz(phiv,Y);
+    x = 1 + (N.*c./(8.*R.*2.*pi)).*A;
+    a= 1-1/x; 
+    
+    if a < 0.5
+        
+    a1(i)=a;
+
+    
+%   Determination of tip speed ratio
+    lambda = y.*(1-a);
+    Lambda(i,1)=lambda;

```

Comment:
You should move `Lambda(i,1)=lambda` inside the `if alpha < alphamax` statement.
With this correction you will solve the problem.

Reply:
The correction has been applied.